runtime/lua/vim/filetype/detect.lua

```diff
@@ -940,7 +940,18 @@ function M.proto(bufnr, default)
   if getlines(bufnr, 2):find('.;$') then
     return 'cpp'
   else
-    return default
+    -- recognize Prolog by specific text in the first non-empty line
+    -- require a blank after the '%' because Perl uses "%list" and "%translate"
+    local line = nextnonblank(bufnr, 1)
+    if
+      line:find('<prolog>')
```

Comment:
`\<` and `\>` are word boundaries in Vim regex. Lua doesn't have an equivalent, so we have to fallback to regex here.

```suggestion
      matchregex(line, [[\<prolog\>]])
```

Reply:
Forgot that upstream just copy-pasted this from the Perl detection -- which is already ported, so I could do the same \o/

(Comment should have tipped me off -- I left it as is, just like upstream.)